platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -57,7 +57,18 @@ Example:
 ADD RELEASE FEATURES BELOW
 vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->
 
+<a name="build-authoring"></a>
+### Build authoring improvements
 
+Gradle provides rich APIs for plugin authors and build engineers to develop custom build logic.
+
+#### Configurations are initialized lazily
+
+Similar to Tasks, Configurations are now initialized lazily, only when necessary.
+
+Now, when applying the `Base` plugin or any plugins derived from it such as the JVM plugins, configurations that are lazily declared using `register` or through the incubating role-based factory methods will no longer be eagerly realized unless required.
+
+Configurations should be declared using the `register` method instead of the `create` method to take advantage of the lazy initialization.
```

Comment:
Can we add a code snippet here? It will be more impactful for people that are glancing at the release notes.